pypeit/sensfunc.py

```diff
@@ -0,0 +1,301 @@
+
+
+    def extrapolate(self, samp_fact=1.5):
+
+        # Create a new set of oversampled and padded wavelength grids for the extrapolation
+        wave_extrap_min = self.out_table['WAVE_MIN'].data * (1.0 - self.par['extrap_blu'])
+        wave_extrap_max = self.out_table['WAVE_MAX'].data * (1.0 + self.par['extrap_red'])
+        nspec_extrap = 0
+        # Find the maximum size of the wavewlength grids, since we want everything to have the same
+        for idet in range(self.norderdet):
+            dwave_data, dloglam_data, resln_guess, pix_per_sigma = wvutils.get_sampling(self.out_table['WAVE'][idet])
+            nspec_now = np.ceil(samp_fact * (wave_extrap_max[idet] - wave_extrap_min[idet]) / dwave_data).astype(int)
+            nspec_extrap = np.max([nspec_now, nspec_extrap])
+        # Create the wavelength grid
+        wave_extrap = np.outer(np.arange(nspec_extrap), (wave_extrap_max - wave_extrap_min)/ (nspec_extrap - 1)) + \
+                      np.outer(np.ones(nspec_extrap), wave_extrap_min)
+        sensfunc_extrap = np.zeros_like(wave_extrap)
+        # Evaluate extrapolated sensfunc for all orders detectors
+        for iorddet in range(self.norderdet):
+            sensfunc_extrap[:, iorddet] = self.eval_sensfunc(wave_extrap[:,iorddet], iorddet)
+
+        self.steps.append(inspect.stack()[0][3])
+        return wave_extrap, sensfunc_extrap
+
+    def splice(self, wave):
+
+        msgs.info('Merging sensfunc for {:d} detectors {:}'.format(self.norderdet, self.par['multi_spec_det']))
+        wave_splice_min = wave.min()
+        wave_splice_max = wave.max()
+        wave_splice, _, _ = coadd1d.get_wave_grid(wave, wave_method='linear', wave_grid_min=wave_splice_min,
+                                                  wave_grid_max=wave_splice_max, samp_fact=1.0)
+        sensfunc_splice = np.zeros_like(wave_splice)
+        for idet in range(self.norderdet):
+            wave_min = self.out_table['WAVE_MIN'][idet]
+            wave_max = self.out_table['WAVE_MAX'][idet]
+            if idet == 0:
+                # If this is the bluest detector, extrapolate to wave_extrap_min
+                wave_mask_min = wave_splice_min
+                wave_mask_max = wave_max
+            elif idet == (self.norderdet - 1):
+                # If this is the reddest detector, extrapolate to wave_extrap_max
+                wave_mask_min = wave_min
+                wave_mask_max = wave_splice_max
+            else:
+                wave_mask_min = wave_min
+                wave_mask_max = wave_max
+            splice_wave_mask = (wave_splice >= wave_mask_min) & (wave_splice <= wave_mask_max)
+            sensfunc_splice[splice_wave_mask] = self.eval_sensfunc(wave_splice[splice_wave_mask], idet)
+
+        self.steps.append(inspect.stack()[0][3])
+
+        return wave_splice, sensfunc_splice
+
+    def show(self):
+        pass
+
+# TODO Add a method which optionally merges sensfunc using the nsens > 1 logic
+
+
+
+class IR(SensFunc):
```

Comment:
Change to `class IRSensFunc(SensFunc)`, and the same for the UVIS one.

Reply:
This requires a bunch of changes to the parset and algorithm name. Is there a problem with just UVIS and IR. I don't think this is going to cause name collisions elswhere in pypeit.

Reply:
I still think it needs to be changed, but we can do it in a different PR.